modules/apps/object/object-service/src/main/java/com/liferay/object/internal/definition/tree/TreeFactoryImpl.java

```diff
@@ -27,7 +30,15 @@
 public class TreeFactoryImpl implements TreeFactory {
 
 	@Override
-	public Tree create(long objectDefinitionId) {
+	public Tree create(long objectDefinitionId) throws PortalException {
+		ObjectDefinition objectDefinition =
+			_objectDefinitionLocalService.getObjectDefinition(
+				objectDefinitionId);
+
+		if (objectDefinition.getRootObjectDefinitionId() == 0) {
+			return null;
```

Comment:
@MarinhoFeliphe what do you think about throwing an exception here? 

So when we use `TreeFactory.create` we wouldn't need to handle the exception AND check if `Tree` is `null`.

Reply:
Agreed, good catch! done [here](https://github.com/liferay-objects/liferay-portal/pull/2458/commits/a5baa33e3c3adfdba69be2dc8391917529668aec)